### PR TITLE
added google tag manager scripts

### LINKED ISSUE #78: Add Google Analytics codes for FE and admin of Public demoshop
I need to add GA for Public demoshop - in the beginning of <head> element add those codes to defined URL addresses.

**https://demo.shopsys.com/  and administration**

<script async src="https://www.googletagmanager.com/gtag/js?id=UA-402396-40"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-402396-40');
</script>

**https://dedemo.shopsys.com/**  

<script async src="https://www.googletagmanager.com/gtag/js?id=UA-402396-41"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-402396-41');
</script>

**https://czdemo.shopsys.com/** 
 
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-402396-42"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-402396-42');
</script>

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| We want to measure peoples interest in demoshop.
|New feature| Yes
|Fixes issues| closes #78 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: src/Shopsys/ShopBundle/Resources/views/Front/Inline/MeasuringScript/googleAnalytics.html.twig [hunk 1]
```diff
@@ -7,4 +7,26 @@
     ga('create', '{{ trackingId }}', 'auto');
     ga('send', 'pageview');
 
-</script>
\ No newline at end of file
+</script>
+
+{% if domainId == 1 %}
```
Comment (vitek-rostislav) [inline src/Shopsys/ShopBundle/Resources/views/Front/Inline/MeasuringScript/googleAnalytics.html.twig]: I am no expert in this field, however, would not the new scripts somehow conflict with the script above? Or these are two different things? (https://www.analyticsmania.com/post/google-tag-manager-vs-google-analytics/)

Then, is not it confusing that GTM and GA are both in `googleAnalytics.html.twig` template? :innocent:

File: src/Shopsys/ShopBundle/Controller/Front/ScriptController.php [hunk 2]
```diff
@@ -48,6 +48,7 @@ public function embedAllPagesGoogleAnalyticsScriptAction()
 
         return $this->render('@ShopsysShop/Front/Inline/MeasuringScript/googleAnalytics.html.twig', [
             'trackingId' => $this->scriptFacade->getGoogleAnalyticsTrackingId($this->domain->getId()),
+            'domainId' => $this->domain->getId(),
```
Comment (vitek-rostislav) [inline src/Shopsys/ShopBundle/Controller/Front/ScriptController.php]: this is connected to my other comment - if GA is not activated, then GTM scripts won't be rendered :confused: (see the `if` above)

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hi, I am a little bit worried about the confusion between GA and GTM. Is it Google analytics or Google tag manager script that you are adding?

File: src/Shopsys/ShopBundle/Resources/views/Front/Inline/MeasuringScript/googleAnalytics.html.twig (hunk 1 above)
Comment (TomasLudvik) [reply src/Shopsys/ShopBundle/Resources/views/Front/Inline/MeasuringScript/googleAnalytics.html.twig]: I think they are not in conflict and I have split them as you expected

File: src/Shopsys/ShopBundle/Controller/Front/ScriptController.php (hunk 2 above)
Comment (TomasLudvik) [reply src/Shopsys/ShopBundle/Controller/Front/ScriptController.php]: You are right, fixed

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
     - domains and domains_url search index routing were added for german locale
     - CategoryRepositoryTest was fixed for third domain
     - translations were copied from en language and prefixed with (de)
+- [#81 - added google analytics scripts](https://github.com/shopsys/demoshop/pull/81)
```
Comment (vitek-rostislav) [inline CHANGELOG.md]:
```suggestion
- [#81 - added google tag manager scripts](https://github.com/shopsys/demoshop/pull/81)
```

File: src/Shopsys/ShopBundle/Controller/Front/ScriptController.php [hunk 4]
```diff
@@ -48,6 +48,14 @@ public function embedAllPagesGoogleAnalyticsScriptAction()
...
```
Comment (vitek-rostislav) [inline src/Shopsys/ShopBundle/Controller/Front/ScriptController.php]: this is not needed here anymore :wink:

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hi, just 2 small things

File: src/Shopsys/ShopBundle/Controller/Front/ScriptController.php (hunk 4 above)
Comment (TomasLudvik) [reply src/Shopsys/ShopBundle/Controller/Front/ScriptController.php]: fixed and rebased

**Review by vitek-rostislav — APPROVED** [review]

**Review by LukasHeinz — APPROVED** [review]

**Pull request merged**